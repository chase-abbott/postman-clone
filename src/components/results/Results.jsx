import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const ResultsArea = styled.pre`
background-color: white;
height: 50vh;
border: 1px solid lightgray;
width: 80%;
overflow: auto;
justify-self: end;
`;


const Results = ({ data }) => {
  return <ResultsArea> {JSON.stringify(data, null, 2) } </ResultsArea>;
};

Results.propTypes = {
  data: PropTypes.array.isRequired
};

export default Results;
