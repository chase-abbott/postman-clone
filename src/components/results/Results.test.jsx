import React from 'react';
import { render } from '@testing-library/react';
import Results from './Results.jsx';

describe('Results component snapshot', () => {
  it('Renders a Results component on the page', () => {
    const data = [{ json: 'json' }];
    const { asFragment } = render(<Results data={data}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
