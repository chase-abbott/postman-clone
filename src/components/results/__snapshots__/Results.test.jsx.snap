// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component snapshot Renders a Results component on the page 1`] = `
<DocumentFragment>
  <pre
    class="sc-bdnxRM qMVbX"
    data-testid="results"
  >
     [
  {
    "json": "json"
  }
] 
  </pre>
</DocumentFragment>
`;
